dev/archery/archery/integration/tester.py

```diff
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Base class for language-specific integration test harnesses
+
+import subprocess
+
+
+class Tester(object):
+    PRODUCER = False
+    CONSUMER = False
+    FLIGHT_SERVER = False
+    FLIGHT_CLIENT = False
+    FLIGHT_PORT = 31337
+
+    def __init__(self, debug=False, **args):
+        self.args = args
+        self.debug = debug
+
+    def run_shell_command(self, cmd):
+        cmd = ' '.join(cmd)
+        if self.debug:
```

Comment:
The `util/command.py` provides this functionality transparently (connected with `--debug`). Might be a big change to refactor all runners to use this facility. Just pointing so you know it exists.

Reply:
Yes will leave to follow up